src/StereoDB/StereoDb.fs

```diff
@@ -122,10 +136,70 @@ type internal StereoDb<'TSchema when 'TSchema :> IDbSchema>(logger: ILogger, sch
             with
                 ex -> ()
     }
+
+    let createHeartbeatTopic (config: HeartbeatConfig) = valueTask {
+        use adminClient = AdminClientBuilder(
+            AdminClientConfig(BootstrapServers = config.KafkaBootstrapServers)).Build()
+
+        try
+            do! adminClient.CreateTopicsAsync(
+                [TopicSpecification(
+                    Name = _heartbeatTopic,
+                    Configs = Dictionary<string, string> (
+                        dict [("retention.ms", config.KafkaHeartbeatTopicRetention.TotalMilliseconds.ToString())]
+                    )
+                )],
+                CreateTopicsOptions()
+            )
+        with
+            ex -> logger.Error(ex, "Error during creating hearbeat topic")
+    }
     
+    let listenHeartbeatTopics (config: HeartbeatConfig) = valueTask {
+        use kafkaConsumer = ConsumerBuilder<string, string>(
+            ConsumerConfig(
+                BootstrapServers = settings.HeartbeatConfig.KafkaBootstrapServers,
+                AutoOffsetReset = AutoOffsetReset.Latest,
+                GroupId = config.KafkaHeartbeatTopicPrefix)).Build()
+
+        kafkaConsumer.Subscribe($"^{config.KafkaHeartbeatTopicPrefix}.+")  // subscribe to all heartbeat topics by regex
```

Comment:
It seems your consumer should listen Heartbeats only from your ClusterId

Reply:
updated